tests/providers/apache/druid/hooks/test_druid.py

```diff
@@ -213,6 +214,36 @@ def test_get_auth_with_no_user_and_password(self, mock_get_connection):
         mock_get_connection.return_value = get_conn_value
         assert self.db_hook.get_auth() is None
 
+class TestDruidDbApiHookConn:
+    def setup_method(self):
+        self.connection = Connection(
+            host="test_host",
+            port=10000,
+            login="test_login",
+            password="test_password",
+            extra='{"endpoint": "/test/endpoint", "schema": "https"}'
+        )
+
+        class UnitTestDruidDbApiHook(DruidDbApiHook):
+            conn_name_attr = "druid_broker_conn_id"
+
+        self.db_hook = UnitTestDruidDbApiHook(context={'query_origin': 'airflow'})
+        self.db_hook.get_connection = MagicMock()
+        self.db_hook.get_connection.return_value = self.connection
```

Comment:
I'm not sure that we need to create new hook for testing purpose. 
You could mock required methods in test cases methods + [`pytest.mark.parametrize`](https://docs.pytest.org/en/latest/how-to/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions)

Reply:
You're right, I refactored the test case as described.